WeatherTests/UnitTests/WeatherServiceTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Moq;
+using NUnit.Framework;
+using System;
+using System.Threading.Tasks;
+using WeatherBL;
+using WeatherBL.Interfaces;
+using WeatherBL.Models;
+using WeatherBL.Validators;
+using WeatherDAL;
+using WeatherDAL.Models;
+
+namespace WeatherTests.UnitTests
+{
+    [TestFixture]
+    public class WeatherServiceTests
+    {       
+        private WeatherService _service;
+        private WeatherService _testService;
+        private WeatherService _testProvider;
+        private Mock<IWeatherProvider> _mockProvider;
+        private Mock<IWeatherProvider> _mockProvide;
+        private Mock<IValidator> _mockValidator;
+        private Mock<IValidator> _mockValidate;
+        private double temp;
+        private string city;
+        private string emptyCity;
+
+
+        public WeatherServiceTests()
+        {
+            temp = 300.00;
+            emptyCity = "";
+            city = "Tashkent";
+            _mockProvider = new Mock<IWeatherProvider>();
+            _mockValidator = new Mock<IValidator>();          
+            _mockProvider.Setup(m => m.GetWeatherAsync(city)).ReturnsAsync(new WeatherResponse() { StatusCode = 200, Main = new Main() { Temp = temp } });            
+            _mockValidator.Setup(m => m.Validate(city)).Returns(true);
+            _service = new WeatherService(_mockProvider.Object, _mockValidator.Object);
+        }
+        
+        [Test]
+        public async Task GetCurrentWeatherAsync_CityNameEqualsResponseCityName_ReturnsAreEqual()
```

Comment:
Read about test methods naming and fix this

Reply:
read